src/utils.js

```diff
@@ -124,6 +124,15 @@ function isSimpleObject(value) {
   return value !== null && value !== undefined && value.toString() === '[object Object]';
 }
 
+function containsHeaders(reference, target) {
+  console.log(reference, target);
+  if (!reference || !target) return deepEqual(reference, target);
+  if (typeof reference !== 'object' || typeof target !== 'object') return deepEqual(reference, target);
```

Comment:
For rear edge case if for any reason `handler[2]` is not object but say a String instance(`'false'`) - so it would not match with headers object `{ 0: 'f' }` even if it's passed in real request.